api/models/app.go

```diff
@@ -474,18 +486,22 @@ func (a *App) RunAttached(process, command, releaseId string, height, width int,
 		return err
 	}
 
-	// If no releaseId is provided, use the last promoted release to run this process
-	// otherwise iterate over the previous revisions (starting with the latest) looking for the releaseId specified.
-	if releaseId == "" {
-		releaseId = a.Release
-
-		for _, container = range task.TaskDefinition.ContainerDefinitions {
-			if *container.Name == process {
-				break
-			}
+	for _, container = range task.TaskDefinition.ContainerDefinitions {
+		if *container.Name == process {
+			break
 		}
+	}
 
-	} else {
+	if container == nil {
+		return fmt.Errorf("unable to find container for %s", process)
+	}
+
+	// If the release ID provided does not equal the active one, some logic is needed to determine the next steps.
+	// - For a previous release, we iterate over the previous TaskDefinition revisions (starting with the latest) looking for the releaseId specified.
+	// - If the release has yet to be promoted, we use the most recent TaskDefinition with the provided release's environment.
+	if releaseId != a.Release {
+
+		var releaseContainer *ecs.ContainerDefinition
```

Comment:
This reminds me of the recent builds cleanup change. We now only retain the last 50 builds and only list 20 by default.  (We also have a rate limit problem in the build list/cleanup here, which is why its on my mind for this review).

So +1 for limiting it to N most recent releases. We only return 20 in the ReleaseList API so that seems like a reasonable number.